### ISSUE TITLE
Android export: Add setting for custom source ZIP for custom builds

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#36728]
**Godot version:**
3.2-stable, master (e2b66ca)

**OS/device including version:**
Exporting to Android from any device.

**Issue description:**
The Android export preset has settings for "Custom Templates", both Release and Debug, like other platforms:
![Screenshot_20200302_145530](https://user-images.githubusercontent.com/4701338/75682587-f21b9d80-5c95-11ea-9490-ce120f363abc.png)

But since 3.2, there's also an `android_source.zip` source ZIP that is used when "Custom Build" is enabled. There's no way to specify a path to a custom source ZIP currently to used with the custom debug and release templates.

**Steps to reproduce:**
- Build custom Android template
- Try to use it with the Custom Build workflow. You are forced to copy release, debug and source custom templates to `~/.local/share/godot/templates/<version>/` or equivalent.

CC @godotengine/android

**Comment by TheMagnat** [comment]
Hello,
My friend and I want to try to take care of this, if no one is already on it.
If anyone has an idea for where to start, don't hesitate!

**Comment by m4gr3d** [comment]
@TheMagnat feel free to take a crack at it. The menu in the screenshot is populated from the [export.cpp](https://github.com/godotengine/godot/blob/master/platform/android/export/export.cpp) file so that would be a good place to get started.

**Comment by horneac** [comment]
Hello,
I'm a beginner and I wanted to try my shot at this for my first contribution since I saw there was still no pull request for this.
But, after i managed to add an entry for specifying the path and checking that path besides the default one, I found out that it isn't used anywhere, because for custom build it already asks to install that build in the project and as I see it the check for the custom template path should be removed as it isn't used (as apposed to adding a way to specify that path as specified in the issue).

I worked on commit b383484e445 (Merge pull request #37421 from akien-mga/python-format-black )

Is this right? if so, can I delete that check and make a pull request for it?

**Comment by SkyLucilfer** [comment]
@akien-mga Just to make sure, you want it to be possible to specify the location of an android_source.zip? 
Then when we click "Install Android Build Template..." in the Project menu, the specified source ZIP will be used to set up the project for custom Android builds ( by installing the source ZIP to res://android/build )

**Comment by akien-mga** [comment]
Yeah that's the idea. It would likely be `custom_template/source` after `custom_template/release`.

**Comment by SkyLucilfer** [comment]
@akien-mga 
I think it might be better to specify the location of the custom source ZIP when we click "Install Android Build Template..." instead of specifying it in an Android export preset.

If we specify it in an Android export preset, there can be multiple Android export presets, each specifying a different custom source ZIP. "Install Android Build Template..." will not know which custom source ZIP to use then. We could use the first occurence, but it gets very messy. Eventually all Android export presets will use the custom build template in "res://android/build" ( after we install it ), so no need to have a custom source ZIP for each.

**Comment by edoblette** [comment]
Hi, my collegue @TheMagnat and myself are working on this issue, what label should we display for the Android Build Template file picker ?

**Comment by Calinou** [comment]
@edoblette "Custom Android Build Template"?

**Comment by TheMagnat** [comment]
We've edited the [export.cpp](https://github.com/godotengine/godot/blob/master/platform/android/export/export.cpp) file to now have a field to specify a path to the android_souce.zip file.
If the path is empty it will use the default android_source.zip template file.
This is for export.cpp but now we have to deal with the installation of the template in ress/android/build. I figured out that this is done in the [export_template_manager.cpp](https://github.com/godotengine/godot/blob/master/editor/export_template_manager.cpp) file,
but in this file, it uses the default template path.
First I would like to know if you think that we are on the right path and if so, if there is a way to access the path filled in one of the parameters of the EditorExportPreset (in the export.cpp file, a new option we added to specify the android_source.zip path).
Here is our [fork](https://github.com/TheMagnat/godot) of the project with the edits we've done for the moment.
